### PR TITLE
Avoid breaking call chains unnecessarily

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR attempts to fix the formatting of the following expression:

```python
max_message_id = (
    Message.objects.filter(recipient=recipient).order_by("id").reverse()[0].id
)
```

Specifically, Black preserves _that_ formatting, while we do:

```python
max_message_id = (
    Message.objects.filter(recipient=recipient)
    .order_by("id")
    .reverse()[0]
    .id
)
```

The fix here is to add a group around the entire call chain.

## Test Plan

Before:

- `zulip`: 0.99702
- `django`: 0.99784
- `warehouse`: 0.99585
- `build`: 0.75623
- `transformers`: 0.99470
- `cpython`: 0.75989
- `typeshed`: 0.74853

After:

- `zulip`: 0.99703
- `django`: 0.99791
- `warehouse`: 0.99586
- `build`: 0.75623
- `transformers`: 0.99470
- `cpython`: 0.75989
- `typeshed`: 0.74853

### CODE REVIEW

File: crates/ruff_python_formatter/src/expression/expr_attribute.rs [hunk 1]
```diff
@@ -35,105 +35,117 @@ impl FormatNodeRule<ExprAttribute> for FormatExprAttribute {
...
         let call_chain_layout = self.call_chain_layout.apply_in_node(item, f);
 
-        let needs_parentheses = matches!(
-            value.as_ref(),
-            Expr::Constant(ExprConstant {
-                value: Constant::Int(_) | Constant::Float(_),
-                ..
-            })
-        );
+        let format_inner = format_with(|f: &mut PyFormatter| {
```
Comment (charliermarsh) [inline crates/ruff_python_formatter/src/expression/expr_attribute.rs:38]: (This is purely indenting the existing body in a `format_with`. No changes in the indented body.)

**Review by MichaReiser — CHANGES_REQUESTED** [review]
Please update your test plan with the similarity index. It is otherwise hard to judge if this is an incorrect special case or indeed the expected behavior

**Comment by charliermarsh** [comment]
Every time I do this manually, a little part of me dies.

**Comment by charliermarsh** [comment]
Done.

**Review by MichaReiser — APPROVED** [review]
Thanks. 

Yeah, the process is cumbersome especially because you need to manually delete text.

File: crates/ruff_python_formatter/src/expression/expr_attribute.rs [hunk 2]
```diff
@@ -35,105 +35,117 @@ impl FormatNodeRule<ExprAttribute> for FormatExprAttribute {
...
-                    leading_comments(leading_attribute_comments),
-                    text("."),
-                    trailing_comments(trailing_dot_comments),
-                    attr.format()
-                ]
-            )
+        if matches!(
+            (self.call_chain_layout, call_chain_layout),
+            (CallChainLayout::Default, CallChainLayout::Fluent)
+        ) {
```
Comment (konstin) [inline crates/ruff_python_formatter/src/expression/expr_attribute.rs:145]: nit: you can use `==` here

File: crates/ruff_python_formatter/resources/test/fixtures/ruff/parentheses/call_chains.py [hunk 3]
```diff
@@ -154,4 +154,6 @@
     five,
 )
 
-
+max_message_id = (
+    Message.objects.filter(recipient=recipient).order_by("id").reverse()[0].id
+)
```
Comment (konstin) [inline crates/ruff_python_formatter/resources/test/fixtures/ruff/parentheses/call_chains.py:159]: This works but i'd like to have it be part of the tests

```suggestion
)

max_message_id = (
    Message.objects.filter(recipient=recipient).order_by("id").reverse()[0].id()
)
```

**Review by konstin — APPROVED** [review]

**Comment by charliermarsh** [comment]
> Yeah, the process is cumbersome especially because you need to manually delete text.

Also because I have to go find another PR on main to find the baseline, right?

**Pull request merged**